data/src/test/java/com/linkedin/data/schema/TestSchemaToPdlEncoder.java

```diff
@@ -50,4 +50,109 @@ public void testEncodeRecordWithEmptyDataMapInProperty() throws IOException
     assertEquals(source.getProperties(), encoded.getProperties());
     assertEquals(source, encoded);
   }
+
+  @Test
+  public void testEncodeSortsNestedPropertyMap() throws IOException
+  {
+    String inputSchema = String.join("\n",
+        "@nested = {",
+        "  \"b\" : \"b\",",
+        "  \"a\" : \"a\"",
+        "}",
+        "record A {}");
+
+    DataSchema schema = TestUtil.dataSchemaFromPdlString(inputSchema);
+
+    String indentedSchema = SchemaToPdlEncoder.schemaToPdl(schema, SchemaToPdlEncoder.EncodingStyle.INDENTED);
+
+    assertEquals(String.join("\n",
+        "@nested = {",
+        "  \"a\" : \"a\",",
+        "  \"b\" : \"b\"",
+        "}",
+        "record A {}"), indentedSchema);
+
+    String compactSchema = SchemaToPdlEncoder.schemaToPdl(schema, SchemaToPdlEncoder.EncodingStyle.COMPACT);
+
+    assertEquals("@nested={\"a\":\"a\",\"b\":\"b\"}record A{}", compactSchema);
+  }
+
+  @Test
+  public void testEncodeDefaultValueFieldsInSchemaOrder() throws IOException
+  {
+    String inputSchema = String.join("\n",
+        "record A {",
+        "",
+        "  b: record B {",
+        "    b1: string",
+        "",
+        "    c: record C {",
+        "      c2: int",
+        "      c1: boolean",
+        "",
+        "      c3: array[string]",
+        "    }",
+        "    b2: double",
+        "  } = {",
+        "    \"b1\" : \"hello\",",
+        "    \"b2\" : 0.05,",
+        "    \"c\" : {",
+        "      \"c1\" : true,",
+        "      \"c2\" : 100,",
+        "      \"c3\" : [ \"one\", \"two\" ]",
+        "    }",
+        "  }",
+        "}");
+
+    DataSchema schema = TestUtil.dataSchemaFromPdlString(inputSchema);
+
+    String indentedSchema = SchemaToPdlEncoder.schemaToPdl(schema, SchemaToPdlEncoder.EncodingStyle.INDENTED);
+
+    assertEquals(String.join("\n",
+        "record A {",
+        "",
+        "  b: record B {",
+        "    b1: string",
+        "",
+        "    c: record C {",
+        "      c2: int",
+        "      c1: boolean",
+        "",
+        "      c3: array[string]",
+        "    }",
+        "    b2: double",
+        "  } = {",
+        "    \"b1\" : \"hello\",",
+        "    \"c\" : {",
+        "      \"c2\" : 100,",
+        "      \"c1\" : true,",
+        "      \"c3\" : [ \"one\", \"two\" ]",
+        "    },",
+        "    \"b2\" : 0.05",
+        "  }",
+        "}"), indentedSchema);
+
+    String compactSchema = SchemaToPdlEncoder.schemaToPdl(schema, SchemaToPdlEncoder.EncodingStyle.COMPACT);
+
+    assertEquals(String.join("",
```

Comment:
can you format this as well, thanks!

Reply:
I added indentation on each line and trim it prior to joining each string. I hope it looks better now.